Add packages for SQLite3 Multiple Ciphers

bricelam opened the pull request:
Depends on ericsink/cb#19

/cc @utelle

bricelam commented on src/global.json:
```diff
@@ -1,4 +1,9 @@
 {
+	"sdk":
+	{
+		"version": "6.0.100",
+		"rollForward": "latestFeature"
+	},
```
This is to make the repo easier to work with locally. It won't use the .NET 8 SDK which throw for net6.0-ios and android targets. It reflects the 6.0.x semantics in the GitHub action.

ericsink replied:
I've been adding something similar to my local work directory.

I'm a little bit tempted to deprecate net6.0 next week.

But net6.0 is still in support for another year.

But the net6.0 mobile targets are a special case.

ericsink replied:
While I'm confessing my temptations, I'd love to deprecate support for .NET Framework too, but that's a whole different can of worms.  :-)

bricelam replied:
In our fork, I compromised by supporting the base net6.0 TFM, but requiring net7.0-android and -ios.

bricelam replied:
I still want .NET Framework support, but only for VS.

ericsink replied:
Speaking broadly, not just about SQLite...

I always wonder how the support story for .NET Framework would be different without VS in the picture.  I assume it'll probably never move to [.NET Core and its descendants].  Lots of enterprise apps are in the same boat, but VS is a special case.

No comments expected -- I'm just musing a bit.

bricelam commented:
Using the binaries from our fork of `cb`, everything builds and passes locally.

bricelam commented:
I’ll update this to use the `e_` prefix to avoid clashes.

ericsink commented:
I've reviewed these diffs and I don't see any obvious problems.

I do think the `e_` prefix for sqlite3mc is the way to go.

The PR for `cb` has been merged and new builds are done, so the next time this PR is checked, I think it'll pass.

bricelam commented:
Updated to prefix with "e_". I also added a provider package without the prefix to enable using the version installed from source (and any future OS packages).

bricelam commented:
> The PR for `cb` has been merged and new builds are done, so the next time this PR is checked, I think it'll pass.

Looks like you still need to merge from `staging` to `master`.

ericsink commented:
Oops!  I thought I did that...

bricelam commented:
```
Failed SQLitePCL.Tests.test_cases.test_sqlite3_soft_heap_limit64
Expected: 20971520
Actual:   10485760
```

ericsink commented:
Hmmm.  And this error wasn't happening when you ran the test suite using the builds you grabbed from the cb fork?

bricelam commented:
I was using the binaries from the `sqlite3mc` fork of https://github.com/utelle/SQLite3MultipleCiphers-cb and didn't get the failure locally.

ericsink commented:
I see reasons to ask questions about how the heap limit functions might have different behavior for sqlite3mc.

But I see nothing to explain why the error wasn't happening for you before.  It's like something is different about the new builds from the merged `cb`.

@utelle

ericsink commented:
I pushed a commit into this PR to disable the failing tests, just for now.

utelle commented:
> I see reasons to ask questions about how the heap limit functions might have different behavior for sqlite3mc.

I have absolutely no idea. What I can say is that sqlite3mc does not actively interfer with the heap limits. sqlite3mc implements a VFS shim which sits between SQLite and the real VFS. If encryption is not enabled it passes the page data through to the real VFS unchanged.

> But I see nothing to explain why the error wasn't happening for you before. It's like something is different about the new builds from the merged `cb`.

I will try to perform this heap limit test directly with sqlite3mc alone, outside of any .NET environment.
